protocol/bin-prot/src/bin_prot_deser.rs

```diff
@@ -0,0 +1,26 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::de::Deserializer;
+use serde::{Deserialize, Serialize};
+
+pub trait BinProtDeser {
```

Comment:
It's not clear as to why we need to create a trait for the methods below? Are tests the only place those methods are used, if yes, I think we can have those methods in tests itself rather than being part of the library. Thoughts?

Reply:
I see what you are trying to do here but I agree we should keep it out of the public API for `bin_prot` and try and keep it to the standard serde api where possible

Reply:
Fixed, renamed to BinProtEncodable under crypto